wasm_src/gsl_wrapper/post.ts

```diff
@@ -129,4 +130,81 @@ Module.savitzkyGolaySmooth = function (xIn: Float64Array | Float32Array, yIn: Fl
+
+    Module.inputData = Module._malloc(componentN * 3 * 8);
+    Module.HEAPF64.set(new Float64Array(inputData), Module.inputData / 8);
+    const inputArray: number[] = [];
+    for (let i = 0 ; i < componentN; i++) {
+        inputArray.push(Module.inputData + i * 3 * 8);
+    }
+    Module.inputArray = Module._malloc(componentN * 4);
+    Module.HEAPU32.set(new Uint32Array(inputArray), Module.inputArray / 4);
+
+    Module.lockedInputData = Module._malloc(componentN * 3 * 4);
+    Module.HEAP32.set(new Int32Array(lockedInputData), Module.lockedInputData / 4);
+    const lockedInputArray: number[] = [];
+    for (let i = 0; i < componentN; i++) {
+        lockedInputArray.push(Module.lockedInputData + i * 3 * 4)
+    }
+    Module.lockedInputArray = Module._malloc(componentN * 4);
+    Module.HEAPU32.set(new Uint32Array(lockedInputArray), Module.lockedInputArray / 4);
+
+    Module.resultOrderValues = Module._malloc(4 * 8); // yIntercept, yInterceptError, slope, slopeError
+    Module.resultAmp = Module._malloc(componentN * 8 * 2); // amp with error
+    Module.resultCenter = Module._malloc(componentN * 8 * 2); // center with error
+    Module.resultFwhm = Module._malloc(componentN * 8 * 2); // fwhm with error
+    Module.resultIntegral = Module._malloc(componentN * 8 * 2); // integral with error
+
+    const log = Module.fittingGaussian(
+        Module.xIn, Module.yIn, dataN,
+        Module.inputArray, Module.lockedInputArray, componentN,
+        fittingFunction, Module.orderValues, Module.lockedOrderValues,
+        Module.resultAmp, Module.resultCenter, Module.resultFwhm, Module.resultOrderValues, Module.resultIntegral);
+
+    const orderValuesOut = new Float64Array(Module.HEAPF64.buffer, Module.resultOrderValues, 4).slice();
+    const centerOut = new Float64Array(Module.HEAPF64.buffer, Module.resultCenter, componentN * 2).slice();
+    const ampOut = new Float64Array(Module.HEAPF64.buffer, Module.resultAmp, componentN * 2).slice();
+    const fwhmOut = new Float64Array(Module.HEAPF64.buffer, Module.resultFwhm, componentN * 2).slice();
+    const integralOut = new Float64Array(Module.HEAPF64.buffer, Module.resultIntegral, componentN * 2).slice()
+
+    Module._free(Module.xIn);
+    Module._free(Module.yIn);
+
+    Module._free(Module.orderValues);
+    Module._free(Module.lockedOrderValues);
+
+    Module._free(Module.inputArray);
+    Module._free(Module.inputData);
+
+    Module._free(Module.lockedInputArray);
+    Module._free(Module.lockedInputData);
+
+    Module._free(Module.resultOrderValues);
+    Module._free(Module.resultCenter);
+    Module._free(Module.resultAmp);
+    Module._free(Module.resultFwhm);
+    Module._free(Module.resultIntegral);
+
+    return {center: centerOut, yIntercept: orderValuesOut[0], yInterceptError: orderValuesOut[1], slope: orderValuesOut[2], slopeError: orderValuesOut[3], amp: ampOut, fwhm: fwhmOut, log: log, integral: integralOut};
```

Comment:
you shouldn't access `orderValuesOut` after you've freed it. Rather create a `result` object first, then free memory and then return `result`

Reply:
fixed.